tensorboard/plugins/graph/tf_graph_common/render.ts

```diff
@@ -235,17 +235,26 @@ export class RenderGraphInfo {
         .domain(memoryExtent)
         .range(PARAMS.minMaxColors);
 
-    // Find also the minimum and maximum compute time.
-    let computeTimeExtent = d3.extent(topLevelGraph.nodes(),
-        (nodeName, index) => {
-      let node = topLevelGraph.node(nodeName);
+    // Find the maximum and minimum compute time in the whole graph.
```

Comment:
Could you add a comment here noting that, furthermore, the total compute time for a metanode is the sum of the compute time of all nodes within it?

Reply:
I thought it's the max compute time among all nodes within it? At least this is what I saw in the graph I tested with.

Reply:
The logic within the `_.each` block below finds the max `getTotalMicros` throughout the graph. In turn, `getTotalMicros` is the sum of compute time for all nodes underneath it (for all group nodes).
https://github.com/tensorflow/tensorboard/blob/efffcbf9024d618589274b971fb3d898ee65bb7f/tensorboard/plugins/graph/tf_graph_common/graph.ts#L574

Come to think, I believe that means we don't need a complete graph traversal at all because top-level metanodes must have greater compute time than any of its children.

Maybe we can ignore metanodes? ie,

```js
if (node.type === NodeType.META) {
  return;
}
```

Reply:
Actually, lets not ignore metanodes. They get colored too.

Reply:
What you can simply do is iterate through the children of the root (the top-level nodes) because group nodes already include the compute times of their children (as before). I think this change could actually be reverted. Sorry, I didn't notice earlier.

Reply:
But we also need the minimum to calculate the extent right? Does top-level graph have this information too?

Reply:
Ah, sorry. I was out for a while. I think the scale should just range from 0 to max for the sake of graphical integrity. Otherwise, the colors might vary a lot, but the range in compute time might actually be very small.

It's similar to the reason that I prefer bar charts to start at 0:
https://flowingdata.com/2015/08/31/bar-chart-baselines-start-at-zero/

Reply:
I don't really have strong opinion as long as it works. I'll change the code to scale from 0.